### PR TITLE
Add an exception for response status code 404

### PR DESCRIPTION
**Pull request by LaurentPetard**

### CODE REVIEW

File: spec/HttpClient/HttpExceptionHandlerSpec.php [hunk 1]
```diff
@@ -73,7 +91,7 @@ function it_throws_bad_request_exception_when_status_code_4xx(
         RequestInterface $request,
         ResponseInterface $response
     ) {
-        $response->getStatusCode()->willReturn(404);
+        $response->getStatusCode()->willReturn(405);
         $response->getReasonPhrase()->willReturn('Not found');
```
Comment (ahocquard) [inline spec/HttpClient/HttpExceptionHandlerSpec.php]: "Not allowed"

File: src/Exception/NotFoundHttpException.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+namespace Akeneo\Pim\Exception;
+
+/**
+ * Exception thrown when a resource has not been found (404).
+ *
+ * @author    Laurent Petard <laurent.petard@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class NotFoundHttpException extends HttpException
```
Comment (ahocquard) [inline src/Exception/NotFoundHttpException.php]: extends `ClientErrorException`. Same thing with 400 and 422 exception please, I did some mistakes...

File: spec/HttpClient/HttpExceptionHandlerSpec.php [hunk 3]
```diff
@@ -52,6 +53,23 @@ function it_throws_bad_request_exception_when_status_code_401(
             ->during('transformResponseToException', [$request, $response]);
     }
 
+    function it_throws_not_found_exception_when_status_code_404(
```
Comment (ahocquard) [inline spec/HttpClient/HttpExceptionHandlerSpec.php]: Could you change the name of the other specs ? `it_throws_bad_request_exception_when_status_code_401` should be `it_throws_unauthorized_request_exception_when_status_code_401`. I did some copy/past I suppose sorry :/

Comment (LaurentPetard) [inline spec/HttpClient/HttpExceptionHandlerSpec.php]: fixed

**Pull request merged**